ext/VoronoiFVMExtendableFEMBaseExt.jl

```diff
@@ -0,0 +1,287 @@
+                continue
+            end
+        end
+
+        # push p1 a little towards the triangle circumcenter
+        # to avoid it being situated on an edge or node of the flowgrid
+        # (to avoid being stuck on an edge if (p2-p1) is on the edge)
+        bp1 .+= interpolate_eps .* (bary - bp1)
+        eval_trafo!(p1, L2G, bp1)
+
+        (λp2min, imin) = findmin(bp2)
+
+        # if λp2min≥0, then p2 is inside icell and we can simply add the
+        # integral across the line segment between p1 and p2 to the result
+
+        # if not, then p2 is outside of icell and we try to determine
+        # pint which is the point where icell intersects the line segment
+        # [p1,p2] - since pint should be on the boundary of icell,
+        # at least one barycentric coordinate (stored in bpint) should
+        # be zero yielding an expression for the line segment parameter t.
+        # this is not necessarily the previous imin and we have to check
+        # all triangle edges for if going towards that edge actually takes us
+        # closer to p2
+
+        if λp2min ≥ -interpolate_eps
+            SI.integrator(summand, Array{Vector{Float64}}([p1, p2]), [bp1, bp2], icell)
+            result += summand
+            break
+        else
+            # calculate intersection point with corresponding edge
+            imin = 0
+            t = 1.0 + interpolate_eps
+            pint .= p1
+            closestimin = 1
+            closestdist = Inf
+            p1_temp .= 0
+            # check if pint takes us closer to p2 and if bpint is inside the cell *and* if we actually moved with pint from p1
+            while (!iscloser(pint, p1, p2, interpolate_eps)) ||
+                  (any(bpint .<= -interpolate_eps) || any(bpint .>= 1 + interpolate_eps)) ||
+                  (norm(bpint - bp1) <= interpolate_eps)
+                imin += 1
+                if imin == 4
+                    imin = closestimin
+                    bpint .= p1_temp
+                    break
+                end
+                t = bp1[imin] / (bp1[imin] - bp2[imin])
+                bpint = bp1 + t * (bp2 - bp1)
+                eval_trafo!(pint, L2G, bpint)
+                if norm(pint - p2) < closestdist && ((all(bpint .>= -interpolate_eps) && all(bpint .<= 1 + interpolate_eps)))
+                    closestimin = imin
+                    closestdist = norm(pint - p2)
+                    p1_temp .= bpint
+                end
+            end
+            eval_trafo!(pint, L2G, bpint)
+            SI.integrator(summand, Array{Vector{Float64}}([p1, pint]), [bp1, bpint], icell)
+            result += summand
+
+            # proceed to next cell along edge of smallest barycentric coord
+            prevcell = icell
+            icell = xFaceCells[1, xCellFaces[facetogo[1][imin], icell]]
+            icell = icell == prevcell ? xFaceCells[2, xCellFaces[facetogo[1][imin], icell]] : icell
+
+            p1 .= pint
+        end
+
+        i += 1
```

Comment:
What is `i`? Does it appear in the `while` loop? If so, give it a proper name :)

Reply:
`i` basically counts the number of steps through individual cells (=triangles) we've gone through in summing the integral through $\left\\{\lambda \\, \mathtt{p1} + (1-\lambda)\mathtt{p2} \\; : \\; \lambda\in[0,1]\right\\}$.

Used to be useful for debugging reasons, but you're right, it doesn't actually serve a purpose anymore.
I'd suggest getting rid of it.